### ISSUE TITLE
isHomogeneous doesn't seem to work for Weyl algebras

### ISSUE THREAD

**Issue by foxjwill** [Macaulay2/M2#638]
Consider the ring defined in M2 by 

```D = QQ[x, dx, WeylAlgebra=>{x=>dx}, Degrees=>{1,-1}]```
 
This is a quotient of the free algebra QQ<X,dX> by the two-sided ideal J generated by dX*X-X*dX-1. Assigning the free algebra the grading induced by deg(X)=1=-deg(dX), one checks that the ideal J is homogeneous. Hence, D, with the given grading, is (ZZ-)graded.

### Problem
Each of the following return false, despite them all being true:

- `isHomogeneous D`
- `isHomogeneous x`
- `isHomogeneous ideal x`

and so on.

**Comment by antonleykin** [comment]
A `WeylAlgebra`  W is not graded, so `isHomogeneous W` returns false.
The general behaviour of `isHomogeneous RingElement` is to return false when `isHomogeneous Ring` is `false`. An alternative behaviour would be to give an error.

**Comment by DanGrayson** [comment]
General Weyl algebras are not homogeneous, but this one is, because of the negative degree assigned to dx.

**Comment by antonleykin** [comment]
@mahrud: Let's fix `rawIsHomogeneous` in the kernel.

**Comment by ggsmith** [comment]
In version 1.19.1, I get slightly different behaviour than reported above.  However, the bug has not been fixed.
```
i1 : D = QQ[x,dx, WeylAlgebra => {x => dx}, Degrees => {1,-1}]

o1 = D

o1 : PolynomialRing, 1 differential variables

i2 : isHomogeneous D

o2 = true

i3 : isHomogeneous x

o3 = false

i4 : degree x

o4 = {1}

o4 : List

i5 : isHomogeneous dx

o5 = false

i6 : degree dx

o6 = {-1}

o6 : List

i7 : isHomogeneous (x*dx)

o7 = false

i8 : degree (x*dx)

o8 = {0}

o8 : List
```

**Comment by mikestillman** [comment]
Fixed in PR #2440.
